models/files/lfda.R

```diff
@@ -0,0 +1,47 @@
+modelInfo <- list(label = "Local Fisher Discriminant Analysis",
+                  library = c("lfda"),
+                  type = "Classification",
+                  grid = function(x, y, len = NULL) {
+                    stop("grid is not available for lfda. ")
+                  },
+                  fit = function(x, y, r = NULL, metric = c("orthonormalized","plain","weighted"),knn = 5, ...) {
+                    theDots <- list(...)
+
+                    if(is.data.frame(x)) x <- as.matrix(x)
+                    if(is.null(r)){
+                      r <- 3
+                      print("Reduced dimension to 3 by default. ")
+                    }
+
+                    metric <- match.arg(metric)
+                    modelArgs <- c(list(x,y,r,metric,knn))
+                    out <- do.call("lfda", modelArgs)
+
+                    out$call <- NULL
+                    out
+                  },
+                  predict = function(modelFit, newdata = NULL, type = "raw", ...) {
```

Comment:
Once you add a predict method to the `lfda` package, you can replace this with:
```{R}
predict = function(modelFit, newdata, submodels = NULL) 
                    predict(modelFit, newdata)$class
```

Or something similar, which will dispatch the predict method from your lfda package.